### DISCUSSION TITLE
Is MPRIS Media Player Support a thing?

### DISCUSSION THREAD

**Discussion by sonich2401** [mikebrady/shairport-sync#1640] [Q&A] [answered]
I assumed that MPRIS support would make it so my media player applets in my desktop environment would accept shairport-sync as a media source but it is not. Does MPRIS support mean I can use the media player applet to change tracks? I would make a issue ticket but I dont want to make a ticket if the program is working as intended. I tried this on XFCE and Plasma
![Screenshot_2023-03-01_21-02-10](https://user-images.githubusercontent.com/34013705/222335928-6f083536-55a1-4e5b-9276-af3e972e138e.png)

I was able to pull the metadata from MPRIS so I know that functionality is working and shairport-sync-mpris-test-client is showing the mpris data coming through. 

My build settings are

4.1.1-alac-OpenSSL-Avahi-ALSA-pa-pw-stdout-pipe-soxr-metadata-mqtt-dbus-mpris-sysconfdir:/etc
Linux linux 5.15.0-60-generic #66~20.04.1-Ubuntu SMP Wed Jan 25 09:41:30 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by mikebrady** [comment, marked as answer]
Thanks for the question. TBH I haven't tried it recently. AFAIK, `MPRIS` media player applets assume the `MPRIS` service is on the `session` bus rather than the `system` bus. IIRC, you'd have to start Shairport Sync as a user service (e.g., experimentally, from a command line window) and also set the `mpris_service_bus` option to `"session"`. Any control that the applet would have would depend on what remote control the source app (e.g. Spotify or the Music app) allows, which is a bit variable.

**Reply by sonich2401** [reply]
That is what I needed. Thanks mate.
